src/datetime/src/components/DatePicker.js

```diff
@@ -1,21 +1,90 @@
-import React from 'react';
-import { mergeDeep } from 'immutable';
-import { DatePicker } from '@atlaskit/datetime-picker';
-import selectStyles from '../../../style/selectStyles';
-
-export const defaultProps = {
-  selectProps: { styles: selectStyles },
-  // @atlaskit/datetime-picker uses date-fns format tokens
-  // locale defaults to 'en-US'
-  dateFormat: 'MM/DD/YYYY',
-  placeholder: 'MM/DD/YYYY',
+// @flow
+import React, { useCallback, useEffect, useState } from 'react';
+import isFunction from 'lodash/isFunction';
+import { DateTime } from 'luxon';
+import { ThemeProvider } from '@material-ui/styles';
+import { KeyboardDatePicker, MuiPickersUtilsProvider } from '@material-ui/pickers';
+
+import LatticeLuxonUtils from './utils/LatticeLuxonUtils';
+import useInputPropsMemo from './hooks/useInputPropsMemo';
+import { latticeMuiTheme } from './styles';
+
+type DateChange = (date :DateTime, value :string | null) => void;
+type Props = {
+  disabled :boolean;
+  format :string;
+  mask :string;
+  onChange :(dateIso :string) => void;
+  placeholder :string;
+  value :string;
+}
+
+const DatePicker = (props :Props) => {
+  const {
+    disabled,
+    format,
+    mask,
+    onChange,
+    placeholder,
+    value
+  } = props;
+
+  const [selectedDate, setSelectedDate] = useState(null);
+  const [lastValidDate, setLastValidDate] = useState(null);
+
+  useEffect(() => {
+    const date = DateTime.fromISO(value);
+    if (date.isValid) {
+      setSelectedDate(date);
+      setLastValidDate(date);
+    }
+    else {
+      setSelectedDate(null);
+      setLastValidDate(null);
+    }
+  }, [value]);
+
+  const inputProps = useInputPropsMemo(lastValidDate, setSelectedDate);
+
+  const handleDateChange = useCallback<DateChange>((date) => {
+    if (isFunction(onChange)) {
+      if (date === null) {
+        onChange();
+        setLastValidDate(null);
+      }
+      if (date && date.isValid) {
+        const dateIso = date.toISODate();
```

Comment:
wait... so `date` is a luxon `DateTime`?

Reply:
Yes! material-ui/pickers rely on @date-io as a wrapper around popular date libraries. This allows the Mui pickers to use a luxon `DateTime` internally! 

The implementation of the `DatePicker` exposed by LUK however attempts to match the prop contract used in atlaskit just so our apps don't blow up. For example, Date/Time/DateTimePicker  still accept string values as inputs and only outputs values as local ISO strings. You should not have to do very much - if anything at all - when upgrading to the new pickers.